### ISSUE TITLE
BICEP - wait for revision deployment success

### ISSUE THREAD

**Issue by sf-declanomalley** [microsoft/azure-container-apps#1011]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is a: (mark with an x)
- [ ] bug report -> please search issues before submitting
- [x] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)

### Issue description
we have recently implemented start-up health probes which check the service has database connectivity before provisioning the container. This is to prevent configuration changes that break the service deployment. However we have noticed that in bicep when deploying the container app , which is connecting to a container registry to deploy the image using managed identity, the devops pipelines reports success then the image fails to provision. 

I'm not sure if this is by design or a bug but our pipelines are recording success when it has not been successful.

### Steps to reproduce
1. create a yaml pipeline to deploy bicep. the bicep contains container app that deploys an image from a container registry
2. run pipeline
3. pipeline returns as successful 

**Expected behavior** [What you expected to happen.]
the pipeline fails

**Actual behavior** [What actually happened.]
the pipeline succeeds 

**Screenshots**         

### Additional context
Ex. Did this issue occur in the CLI or the Portal?

**Comment by howang-ms** [comment]
Can you double-check is your revision is showing a healthy status after deployment? If it is showing healthy, it means the probes didn't faili
Also, if your start-up health probes failed, you should be able to see this from the ContainerAppSystemLogs.

**Comment by NateB2** [comment]
Yes, experiencing this same issue.  The startup probe fails and the revision shows as unhealthy, but the pipeline succeeds. We do see associated log entries, so it's definitely failing.

The expected behavior should be that the deployment remains "in-progress" until the startup probe returns a response.

**Comment by sf-declanomalley** [comment]
the container app is showing as failed to provision with the bicep pipeline returning success.

**Comment by njuCZ** [comment]
Hi @sf-declanomalley, I am not very clear about your question. Do you mean the containerapp show as succeed while the revision is shown as failed? If this is the case, then it's by design. 

probe take effect on revision. When there is a new revision is created, the containerapp create operation will return as succeed, we don't guarantee the revision is succeed. Customer need to check the revision status themselves

**Comment by njuCZ** [comment]
@confused-cullena Do you mean using bicep to check the status of revision? I am sorry that I am not familiar with bicep tool. I can only introduce some basic idea:
1. check the new revision status
2. each revision has a separate endpoint, call that endpoint
3. for single revision mode, only new revision starts succeed, we will then deactivate old revisions, so you could also check old revision status
